plugins/linux-pipewire/pipewire.c

```diff
@@ -1373,6 +1374,13 @@ void obs_pipewire_stream_video_render(obs_pipewire_stream *obs_pw_stream,
 	gs_blend_state_push();
 	gs_blend_function(GS_BLEND_ONE, GS_BLEND_INVSRCALPHA);
 
+	enum gs_color_format format = s_get_color_space() == GS_CS_SRGB
```

Comment:
Instead of doing this set `OBS_SOURCE_SRGB` capabilities, and and then configure framebuffers as srgb so decompression is done in hardware, you can look at xshm for an example.

Also the assumption that 10bit formats are srgb seems pretty flawed based on windows examples of how programs actually work. Please check if pipewire reports colorspace information so that we can make an informed choice here or if they are lack this.

Reply:
I've implemented the SRGB capabilities and ajusted the code, however I do still have a question.

PipeWire returns `SPA_VIDEO_COLOR_MATRIX_` as color matrix, which in `video_colorspace_from_spa_color_matrix` in pipewire.c:614 is converted into an enum either DEFAULT, 601 or 709. The HDR formats here are simply being passed through as DEFAULT, however seeing how the rest of the code doesn't seem to support HDR anyways I will glance over this detail.
Now what I'm asking is, how do I render textures in this color format. Assuming 10-bit formats are sRGB seems flawed, I agree, however I don't know how I'm suppose to handle 601 and 709 textures. (Do I have to tonemap them from their source format into the current obs space? Which effect technique would I use for this?)

Reply:
We only need to bring the HDR textures into linear space, no tone mapping needs to be done in the plugin. For non-srgb source there are some more draw techniques for common color spaces you can look at the windows game capture plugin for a more complete HDR implementation.

Reply:
@kkartaltepe
got any updates on this?

Reply:
Sorry I thought you had wanted to try and extend this to supporting the other spaces. As its it seems ok, the regular draw technique should work now that you have added srgb tagging to the images maybe confirm that and we can remove the extra technique handling too.

Reply:
> Sorry I thought you had wanted to try and extend this to supporting the other spaces. As it is seems ok, the regular draw technique should work now that you have added srgb tagging to the images maybe confirm that and we can remove the extra technique handling too.

It's always using DrawSrgbDecompress, there aren't any other techniques being used. Replacing DrawSrgbDecompress with Draw doesn't work, if that's what you meant, the textures get the white tint again.

Reply:
@kkartaltepe Mind taking a look at this again?